File: linux/ardium_source/presentation/03_Structs/structs.py
# Topic: Structs
# Define Vector2 and Dot Product

class Vector2:
    def __init__(self, x, y):
        self.x = x
        self.y = y

def dot(a, b):
    return (a.x * b.x) + (a.y * b.y)

def main():
    v1 = Vector2(10, 20)
    v2 = Vector2(5, 5)

    result = dot(v1, v2)
    print(f"Dot Product: {result}")

if __name__ == "__main__":
    main()

File: linux/ardium_source/presentation/03_Structs/structs.rs
// Topic: Structs
// Define Vector2 and Dot Product

struct Vector2 {
    x: i32,
    y: i32,
}

fn dot(a: &Vector2, b: &Vector2) -> i32 {
    return (a.x * b.x) + (a.y * b.y);
}

fn main() {
    let v1 = Vector2 { x: 10, y: 20 };
    let v2 = Vector2 { x: 5, y: 5 };

    let result = dot(&v1, &v2);
    println!("Dot Product: {}", result);
}
